client/src/unifycr-sysio.c

```diff
@@ -356,31 +409,90 @@ int UNIFYCR_WRAP(__xstat)(int vers, const char *path, struct stat *buf)
         }
 
         /* get meta data for this file */
-        unifycr_fid_stat(fid, buf);
+        ret = unifycr_fid_stat(fid, buf);
```

Comment:
`unifycr_fid_stat()` returns `-1` when it fails to find the metadata. Although it is not very likely to happen, still we would need to set `errno` when we return `-1` to caller.